### PR TITLE
Picking endpoint from the properties file and environment variables

### PR DESCRIPTION
**Pull request by ashwanthkumar**
Helps seemless switching between public github and enterprise without any code changes.

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GitHubBuilder.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ public static GitHubBuilder fromCredentials() throws IOException {
     
     }
     
-    public static GitHubBuilder fromEnvironment(String loginVariableName, String passwordVariableName, String oauthVariableName) throws IOException {
+    public static GitHubBuilder fromEnvironment(String loginVariableName, String passwordVariableName, String oauthVariableName, String endpointVariableName) throws IOException {
```
Comment (KostyaSha) [inline src/main/java/org/kohsuke/github/GitHubBuilder.java]: Will broke backward compatibility

File: src/main/java/org/kohsuke/github/GitHubBuilder.java [hunk 2]
```diff
@@ -89,7 +89,11 @@ public static GitHubBuilder fromEnvironment(String loginVariableName, String pas
     	
     	if (oauthValue != null)
     		env.put("oauth", oauthValue);
-    	
+
+        Object endPoint = System.getenv(endpointVariableName);
```
Comment (KostyaSha) [inline src/main/java/org/kohsuke/github/GitHubBuilder.java]: Wrong indentation, and why Object?

Comment (ashwanthkumar) [inline src/main/java/org/kohsuke/github/GitHubBuilder.java]: Sorry about the wrong indentation, will fix it. All `System.getenv` was taken in as Object, that's why. Do I change all to String?

File: src/main/java/org/kohsuke/github/GitHubBuilder.java (hunk 1 above)
Comment (ashwanthkumar) [inline src/main/java/org/kohsuke/github/GitHubBuilder.java]: Ah, Will overload it.

**Pull request merged**